docs/cartesian-product.adoc

```diff
@@ -119,6 +121,110 @@ To demonstrate with a table:
 | 6th test  | 4                 | "B"
 |===
 
+=== `@CartesianEnumSource`
+
+`@CartesianEnumSource` provides a convenient way to use `Enum` constants.
+
+[source,java]
+----
+@CartesianProductTest
+@CartesianEnumSource(ChronoUnit.class)
+void testWithEnumSource(TemporalUnit unit) {
+    assertNotNull(unit);
+}
+----
+
+Like `@CartesianValueSource`, `@CartesianEnumSource` is an annotation that might look familiar - it mimics JUnit's `@EnumSource`, except `@CartesianEnumSource` is repeatable.
+It also does NOT work with `@ParameterizedTest`.
+https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-EnumSource::[You can read about `@EnumSource` here.]
+
+The annotation’s `value` attribute is optional.
+When omitted, the declared type of the first method parameter is used.
```

Comment:
Interesting. Consider the following test:
```java
@CartesianProductTest
@CartesianValueSource(ints = { 1, 2, 3 })
@CartesianEnumSource
void moreParams(int i, ChronoUnit cUnit) {
  // some test code
}
```

Would that work? Would you expect it to work?

Reply:
Uhm, good point. I try to see what happens with a new test case.

Reply:
This also requires aligning the implementation with the latest JUnit 5.7

Reply:
I think I need to be a bit creative to keep this feature as JUnit specifically refers to the first parameter of the test method (which makes sense in the context of `@ParameterizedTest`). How I would like to see it is that the type is inferred based on the position of the annotation. E.g., having:

```java
@CartesianProductTest
@CartesianValueSource(ints = { 1, 2, 3 })
@CartesianEnumSource
@CartesianValueSource(ints = { 4, 5, 6 })
@CartesianEnumSource
void moreParams(int i, ChronoUnit cUnit, int j, MyEnum myEnum) {
  // some test code
}
```

the first `@CartesianEnumSource` should provide all the values of `ChronoUnit`, while the second should provide all the values of `MyEnum`.

Let's see if it's feasible.

Reply:
@Michael1993 I spent some time on this topic and the change is not really straightforward. I propose to finalize this PR without the auto-detection feature and then having a follow-up issue to do this investigation. What do you think?

Reply:
Closing this thread for now. Will open an issue about your proposed improvements.